types/events.go

```diff
@@ -74,23 +81,36 @@ func (em *EventManager) EmitTypedEvents(tevs ...proto.Message) error {
 
 // TypedEventToEvent takes typed event and converts to Event object
 func TypedEventToEvent(tev proto.Message) (Event, error) {
-	evtType := proto.MessageName(tev)
+	// Replace "." in event type with "-" to fix tm event query issue
+	evtType := strings.ReplaceAll(proto.MessageName(tev), ".", "-")
 	evtJSON, err := codec.ProtoMarshalJSON(tev, nil)
 	if err != nil {
 		return Event{}, err
 	}
 
-	var attrMap map[string]json.RawMessage
+	var attrMap map[string]interface{}
 	err = json.Unmarshal(evtJSON, &attrMap)
```

Comment:
Is there something simpler we could do here rather than marshaling and then unmarshalling back again?